### DISCUSSION TITLE
If I import users with a CSV file, which table should I update?

### DISCUSSION THREAD

**Discussion by Daniel-HyunHo-Shin** [supabase/supabase#28312] [Questions] [answered]
Thank you for checking out this question.

I am working on migrating one of my Amplify projects and need to migrate user data by importing a CSV file into the database.

**My question is: To which table should I insert the CSV data?**
**Is it sufficient to insert it only into the auth.user table, or are there other tables that I also need to populate?**

All of our users log in using OAuth2(Google, facebook, apple), so I don't need to worry about passwords in this case.

Thank you for your assistance.

--- Edit --- 
This may not be the most elegant solution, but it should be effective.

Using OAuth2, newly created users will have the same email they used in Amplify.

I can import user data into a temporary table and set a trigger to move that data to the new user table whenever a user with the same email is created.

Alternatively, instead of using a trigger, I could create an import button to handle the data import manually.

While this solution may lack elegance, it is straightforward and easy to implement.

**Comment by GaryAustin1** [comment, marked as answer]
Normally in the code I've seen users use for creating auth users you have to populate the auth.users and auth.identities table.
There are several threads like this here:
https://github.com/orgs/supabase/discussions/5248#discussioncomment-9981648

**Reply by Daniel-HyunHo-Shin** [reply]
Thank you so much!
